packages/app-store/zohocalendar/api/callback.ts

```diff
@@ -64,7 +64,36 @@ async function getHandler(req: NextApiRequest, res: NextApiResponse) {
     expires_in: Math.round(+new Date() / 1000 + responseBody.expires_in),
   };
 
-  await createOAuthAppCredential({ appId: config.slug, type: config.type }, key, req);
+  const credential = await prisma.credential.create({
+    data: {
+      type: config.type,
+      key,
+      userId: req.session.user.id,
+      appId: config.slug,
+    },
+  });
+
+  const calendarResponse = await fetch("https://calendar.zoho.com/api/v1/calendars", {
+    method: "GET",
+    headers: {
+      Authorization: `Bearer ${key.access_token}`,
+      "Content-Type": "application/json",
+    },
+  });
+  const data = await calendarResponse.json();
+
+  const primaryCalendar = data.calendars.find((calendar: any) => calendar.isdefault);
```

Comment:
`nitpick`: Please ensure a type, even if `{ isdefault: boolean }`

Reply:
Uncertain types aren't a good idea, `any` is actually better as it conveys something is really "unknown" - I did mark this as a nitpick as this is a potential issue.

The right fix for an issue like this is to parse the JSON response and validate the isdefault key is there using zod, then you don't need to type any - then throw a descriptive error when this fails the parsing.